Extend arm-resource-patch rule with PATCH body schema diagnostics (#4381)
Azure/typespec-azure: .chronus/changes/arm-patch-body-schema-2026-05-05-02-30-00.md

---
changeKind: feature
packages:
  - "@azure-tools/typespec-azure-resource-manager"
---

Extend the `arm-resource-patch` linter rule to validate PATCH request body schemas. New diagnostics are emitted when:

> - A PATCH request body property is required and the resource property it maps back to has any visibility other than `Lifecycle.Read` by itself (required properties are only allowed when the source resource property is purely read-only, in which case they are filtered out by visibility transforms).

Reviewer: ```suggestion
- A PATCH request body property is required and not read-only.
```

- A PATCH request body property has a default value (defaults are never applied to PATCH requests).

> - A PATCH request body property maps back to a resource property that is read-only or otherwise not updateable (e.g. `@visibility(Lifecycle.Read)`).

Reviewer: This line is no longer valid

> - A PATCH request body property maps back to a resource property whose visibility must either include `Lifecycle.Update` or include `Lifecycle.Read` without `Lifecycle.Create`; other visibility combinations (e.g. `@visibility(Lifecycle.Create)` or `@visibility(Lifecycle.Create, Lifecycle.Read)`) are not allowed. The check is applied recursively to nested model and `Record<Model>` properties. Properties with default visibility, `@visibility(Lifecycle.Update)` (alone or in any combination), or `@visibility(Lifecycle.Read)` without `Lifecycle.Create` are allowed.

Reviewer: ```suggestion
- A PATCH request body property is not updateable and not read-only.
```
